rest_registration/utils/verification.py

```diff
@@ -49,6 +55,31 @@ def build_default_template_context(
     return context
 
 
+def build_default_template(
+    user: 'AbstractBaseUser',
+    template_config_data: Dict[str, Any],
```

Comment:
I think that function should be more general. In my opinion it should receive:

*   ``user`` - the user which is to be notified.
*   ``notification_type`` - value of ``rest_registration.notifications.enums.NotificationType`` enum.
*   ``notification_method`` - value of ``rest_registration.notifications.enums.NotificationMethod``; will be hardcoded for `EMAIL` for now in the caller function
*   ``template_config_data`` - about this one I'm unsure, but as it depends on `request` which will be unavailable here so it should probably be included as well.

Of course, docs for the setting need to be updated as well.

Reply:
This function receives `template_config_data` which is a generic dict generated in `_get_email_template_config_data` using function set with `VERIFICATION_TEMPLATES_SELECTOR`.
`template_selector` already receives `request`, `user`, `notification_method` and `notification_type` (here hardcoded). I think it is redundant to pass the same data here. If one really wanted to access eg. `request`, he could override selector function to pass additional data into the `template_config_data`.

Reply:
That argument makes sense. But in that case, my question is why we need to provide `user` here? Information about the user can be encoded in `template_config` and/or in `context` (as both can be controlled via `VERIFICATION_TEMPLATES_SELECTOR` and `VERIFICATION_TEMPLATE_CONTEXT_BUILDER`), so I'm not sure if that one is not redundant as well? I don't see `user` being used below.

Reply:
I tried to follow a pattern where most functions got a `user` argument, but you are right:
- `VERIFICATION_TEMPLATES_SELECTOR` can select/generate correct template data based on the `user`
- `VERIFICATION_TEMPLATE_CONTEXT_BUILDER` can create correct context with `user` details
I'll remove this argument.